### PR TITLE
Fix 1686 grid margins go off screen

### LINKED ISSUE #1686: New grid layout spacing can go off-screen
### Description

When using pull to refresh of the podcast grid layout, if you pull again before the refresh is complete, the spacing of the grid changes and pushes some tiles off the edge of the screen. Only way to restore is to change the grid layout to another option and then back.

@SergioEstevao I suspect this bug may be related to the changes in https://github.com/Automattic/pocket-casts-ios/pull/1628

### Step-by-step reproduction instructions

1. Go to Podcasts tab
2. Select grid layout
3. Pull to refresh
4. Quickly pull down again, while refresh is still in progress
5. Spacing between tiles will increase, pushing some tiles off the edge.

### Screenshots or screen recording

https://github.com/Automattic/pocket-casts-ios/assets/25402814/5b799caf-6ba4-4f86-8672-cdb2291d550d

### Did you search for existing bug reports?

- [X] I have searched for existing bug reports.

### Device, Operating system, and Pocket Casts app version

iPhone 14 Pro, iOS 17.4.1, Pocket Casts 7.63.0.1

### PR DESCRIPTION
**Pull request by SergioEstevao**
Fixes #1686 

This PR updates the search bar code and mini-player code to ensure it does not change any existing horizontal insets when updating the insets.

## To test

 - Start the app
 - On podcasts view pull to refresh, and while the pull to refresh is ongoing pull to refresh again.
 - Check that the horizontal margins on the grid layout are kept
 - Start the playing any podcast to see the mini-player
 - Check if you have a bottom margin at the end of scroll the podcasts view
 - Tap on the mini-player artwork and choose `Close and Clear Up Next`
 - Check that the bottom at the end of the scroll is still there and horizontal margins are kept

## Checklist

- [x] I have considered if this change warrants user-facing release notes and have added them to `CHANGELOG.md` if necessary.
- [x] I have considered adding unit tests for my changes.
- [x] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/Automattic/pocket-casts-ios/milestone/46">7.63</a>. The due date for this milestone has already passed.<br />
Please assign it to a milestone with a later deadline or check whether the release for this milestone has already been finished.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by leandroalonso — CHANGES_REQUESTED** [review]
This needs to target `trunk`.

**Review by leandroalonso — CHANGES_REQUESTED** [review]
@SergioEstevao I don't believe `trunk` has the grid changes applied. We probably need to re-add them, merge them first, and then update this PR.

Edit: or maybe we could use this one? Anyways, there's some merge conflict.

**Comment by SergioEstevao** [comment]
I will pull the other one before, because this fix only makes sense when you have horizontal margins to test.

**Comment by SergioEstevao** [comment]
@leandroalonso ready for another look now that we re-added the new grid layout to trunk.

File: podcasts/SceneHelper.swift [hunk 1]
```diff
@@ -26,7 +26,8 @@ class SceneHelper {
     class func rootViewController() -> UIViewController? {
         guard !FeatureFlag.newPlayerTransition.enabled else {
             let appScene = connectedScene()?.windows.first(where: { $0.rootViewController is MainTabBarController })
-            return appScene?.rootViewController?.topMostPresentedViewController
+            let rootVC = appScene?.rootViewController
```
Comment (SergioEstevao) [inline podcasts/SceneHelper.swift]: These changes were not part of this PR, This comes from [another PR](https://github.com/Automattic/pocket-casts-ios/pull/1653) that is still not merged in trunk.

Comment (leandroalonso) [reply podcasts/SceneHelper.swift]: Just a `merge trunk` missing. :)

Gone already.

**Review by leandroalonso — APPROVED** [review]

**Pull request merged**